data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3ObjectWorker.java

```diff
@@ -34,8 +37,13 @@
 class S3ObjectWorker {
     private static final Logger LOG = LoggerFactory.getLogger(S3ObjectWorker.class);
     static final String S3_OBJECTS_FAILED_METRIC_NAME = "s3ObjectsFailed";
+    static final String S3_OBJECTS_FAILED_NOT_FOUND_METRIC_NAME = "s3ObjectsNotFound";
+    static final String S3_OBJECTS_FAILED_NOT_FOUND_ACCESS_DENIED = "s3ObjectsAccessDenied";
     static final String S3_OBJECTS_SUCCEEDED_METRIC_NAME = "s3ObjectsSucceeded";
     static final String S3_OBJECTS_TIME_ELAPSED_METRIC_NAME = "s3ObjectReadTimeElapsed";
+    static final String S3_OBJECTS_SIZE = "s3ObjectSizeBytes";
+    static final String S3_OBJECTS_SIZE_PROCESSED = "s3ObjectProcessedBytes";
+    static final String S3_OBJECTS_RECORDS = "s3ObjectsRecords";
```

Comment:
Should we call these Records or Events? This metric is for the number of Events that are extracted from the s3Objects right?

Reply:
That's a good question. We use the term "records" in the buffer metrics and the `opensearch` sink metrics. So I followed that convention. But, we also plan to remove the `Record` class in favor of `Event`.

Reply:
I'd be inclined to calling them `Events` since the `Record` class will be removed in the future.

Reply:
Even right now Record feels like an internal concept that users don't really care about. In our documentation we have been using Event instead of Record

Reply:
Good points. I'll update this.

Reply:
I just pushed a change which renames this metric to `s3ObjectsEvents`.